src-java/base-topology/base-messaging/src/main/java/org/openkilda/messaging/info/switches/v2/LogicalPortsValidationEntryV2.java

```diff
@@ -60,8 +60,12 @@ static LogicalPortsValidationEntryV2 join(List<LogicalPortsValidationEntryV2> en
         builder.proper(joinLists(nonNullEntries.stream().map(LogicalPortsValidationEntryV2::getProper)));
         builder.missing(joinLists(nonNullEntries.stream().map(LogicalPortsValidationEntryV2::getMissing)));
         builder.misconfigured(joinLists(nonNullEntries.stream().map(LogicalPortsValidationEntryV2::getMisconfigured)));
-        nonNullEntries.stream().filter(e -> StringUtils.isNoneBlank(e.error)).findFirst()
-                .map(LogicalPortsValidationEntryV2::getError).ifPresent(builder::error);
+        if (nonNullEntries.stream().map(e -> e.error).distinct().count() > 1) {
+            builder.error(nonNullEntries.stream().map(e -> e.error)
+                    .distinct().collect(Collectors.joining(",")));
+        } else {
+            nonNullEntries.stream().map(e -> e.error).filter(Objects::nonNull).findFirst().ifPresent(builder::error);
```

Comment:
This if-else seems redundant now. joining on a stream of one element returns you simply that element. What you probably wanted to handle here is an empty stream after the filtering to not add an empty string as an error? As an option, you can simplify it by collecting 64-65 into a variable and pass it to the builder if it is not blank (you don't need to collect the stream multiple times). Or use a different reduce operation, if you prefer one-liners: 
```
nonNullEntries.stream().map(e -> e.error).filter(Objects::nonNull).reduce(String::concat).ifPresent(builder::error);
```

Reply:
Thank you, what I actually want to handle in the else statement is the case where only one element exist or none elements at all. 

 `joining on a stream of one element returns you simply that element.` ->
 I might be wrong, but I do not really understand where I am trying to join a stream of one element.. 


I understand the point in your suggested line of code
`nonNullEntries.stream().map(e -> e.error).filter(Objects::nonNull).reduce(String::concat).ifPresent(builder::error);`. 
But here is the trick: If you somehow receive null element and none-null element in the same stream, then according to the filter predicate you will just filter null elements which is not the behaviour I want to receive.  
 What I really want is to keep null element if there is another none-null element exist. And keep this "null" word in the error field at the end. Because this should mean that something wrong with the code that generated this elements..